### DISCUSSION TITLE
Windows above bar in a specific workspace

### DISCUSSION THREAD

**Discussion by RiccardoPP** [hyprwm/Hyprland#2528] [General]
Hello and thanks for Hyprland. 

I'm looking for an "Hyprland  way" to hide Waybar in a specific workspace, or to bring a window above the bar in a specific workspace.

I've a "workspace rule" (thanks for [that](https://github.com/hyprwm/Hyprland/discussions/1345#discussion-4736910)!) to remove gaps, animations, borders and decoration but the bar is still present. A solution could be to assign the overlay layer to a specific window so it could be stay above all the other elements.

I've tried several ways with no success. Right now I've a keybind to toggle/kill Waybar on request but it's not an elegant solution.

Any other ideas? Thanks.

**Comment by NikkSaan** [comment]
At this moment the "hyprland way" to do this is to use a script.
It's still not the most elegant solution but at least it hides/shows waybar automatically.
Depending on your set-up it may not work well or reliably. 
Test it and if you have questions or you want me to modify the behaviour, ask.

To set which workspace you want to hide waybar on set the script variable `hidden_on_workspace` equal to the workspace number. 
To make sure it works you can test it in the terminal first.

auto_hide_waybar.sh

```sh

#!/usr/bin/sh

# change this to the workspace number you want waybar to be hidden on
hidden_on_workspace=2

function hide {
  killall -SIGUSR1 waybar && touch /tmp/waybar_is_hidden
}

function show {
  if [ -f /tmp/waybar_is_hidden ]; then
    killall -SIGUSR1 waybar && rm /tmp/waybar_is_hidden >> /dev/null 2>&1
  fi
}

# give waybar time to start
sleep 1

# if this script is started on the workspace waybar should be hidden on, hide waybar.
current_workspace=$( hyprctl activeworkspace | awk 'match($0, /ID ([0-9]+)/, id){print id[1]}' )
[ $hidden_on_workspace -eq $current_workspace ] && hide

# lisen for events
socat -u "UNIX-CONNECT:/tmp/hypr/$HYPRLAND_INSTANCE_SIGNATURE/.socket2.sock" STDOUT | while read -r event 
do
  case $event in 
    workspace\>\>$hidden_on_workspace ) hide ;;
    workspace\>\>*                    ) show ;;
  esac
done

# show waybar if the scirpt is stopped
show

```

To launch it automatically add an `exec-once` with the script path **after** the ones that launch waybar, like so..

hyprland.conf
```sh
exec-once = waybar
exec-once = auto_hide_waybar.sh
```

ps. Your videos have been a huge source of inspiration and entertainment, thank you!!

**Reply by goofygoober2872** [reply]
is there a way to do this same thing for niri?

**Comment by mintyleaf** [comment]
I have a similar non-critical issue

I heavily used to the special workspace, which is toggled by touchpad gesture with slide up animation and behaving like an overlay workspace floating from bottom
That workspace successfully darkens my (way)bar at the bottom, but i really want to this workspace completely ignore that bar (like when waybar have disabled exclusive flag), laying simply on the top and not wasting space at the bottom

The problem is if i use SIGUSR1 signal to toggle waybar, or replace exclusive flag and reload waybar - windows on the main workspaces will shrink back to toggled waybar which is looking kinda ugly

There is need to be some kind of control of how layers behave on specific workspaces
